apps/ledger-live-mobile/src/components/CustomImage/injectedCode/imageProcessing.ts

```diff
@@ -0,0 +1,208 @@
+declare global {
+  interface Window {
+    ReactNativeWebView: any;
+    processImage: (imgBase64: string) => void;
+    setImageContrast: (val: number) => void;
+    setAndApplyImageContrast: (val: number) => void;
+    requestRawResult: () => void;
+  }
+}
+
+/**
+ * This function is meant to be stringified and its body injected in the webview
+ * It allows us to have access to APIs that are available in browsers and not
+ * in React Native (here Canvas & Image).
+ * */
+function codeToInject() {
+  /**
+   * The following line is a hermes directive that allows
+   * Function.prototype.toString() to return clear stringified code that can
+   * thus be injected in a webview.
+   *
+   * ⚠️ IN DEBUG this doesn't work until you hot reload this file (just save the file and it will work)
+   * see https://github.com/facebook/hermes/issues/612
+   *  */
+
+  "show source";
+
+  const DEBUG = false;
+
+  function clampRGB(val: number) {
+    return Math.min(255, Math.max(val, 0));
+  }
+
+  function contrastRGB(rgbVal: number, contrastVal: number) {
+    return (rgbVal - 128) * contrastVal + 128;
+  }
+
+  // simutaneously apply grayscale and contrast to the image
+  function applyFilter(
+    imageData: Uint8ClampedArray,
+    contrastAmount: number,
+  ): { imageDataResult: Uint8ClampedArray; hexRawResult: string } {
```

Comment:
Could be interesting to have some unit test on `applyFilter`: to have examples of what to expect for some random `imageData` array with a contrast. If there is any edge cases ? (imageData all 0 or 1 ?).

Reply:
I was planning on rather implementing e2e tests (when they're available on LLM) to directly validate the result on given images.

Reply:
yeah good idea :)